src/Components/ProductOverview/ProductInfo/OrderSelection.jsx

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { buildOrder, availableStock } from '../../../Shared/HelperFunctions';
+import { updateSelectedQuantity, updateCart, updateSelectedSize } from '../../../data/actions/productDataAction';
+
+export class OrderSelection extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showCartError: false,
+    };
+    this.showCartError = this.showCartError.bind(this);
+    this.hideCartError = this.hideCartError.bind(this);
+  }
+
+  showCartError(e, selectedSize) {
+    e.preventDefault();
+    if (selectedSize === null) {
+      this.setState({ showCartError: true });
+    }
+  }
+
+  hideCartError(e) {
+    e.preventDefault();
+    this.setState({ showCartError: false });
+  }
+
+  resetQuantity(availableStock, selectedQuantity, updateSelectedQuantity) {
+    if (availableStock < selectedQuantity) {
+      updateSelectedQuantity(1);
+    }
+  }
```

Comment:
Resets selected quantity if available stock for the newly selected size is less the the currently selected quantity